src/cast_core/src/CommitLatency.cpp

```diff
@@ -0,0 +1,227 @@
+          repeat{phaseContext.get<IntegerSpec>("Repeat")},
+          threads{phaseContext.get<IntegerSpec>("Threads")},
+          amountDistribution{-100, 100},
+          options{ExecutionStrategy::getOptionsFrom(phaseContext, "ExecutionsStrategy")} {
+              // write_concern
+              if( phaseContext.get<bool, false>("WriteConcernMajority").value_or(false) ) {
+                  wc.majority(std::chrono::milliseconds{0});
+              }
+              else {
+                  wc.nodes( (std::int32_t) phaseContext.get<IntegerSpec, false>("WriteConcern").value_or(1) );
+              }
+              if( phaseContext.get<bool, false>("WriteConcernJournal").value_or(false) ) {
+                  wc.journal(true);
+              }
+              if( ! phaseContext.get<bool, false>("WriteConcernJournal").value_or(true) ) {
+                  // On some mongod versions it makes a difference whether journal is unset vs explicitly set to false
+                  wc.journal(false);
+              }
+              if ( useTransaction ) {
+                  optionsTransaction.write_concern(wc);
+              }
+              else {
+                  optionsUpdate.write_concern(wc);
+              }
+
+              // read_concern
+              rc.acknowledge_string(phaseContext.get<std::string>("ReadConcern"));
+              // Ugh... read_concern cannot be set for operations individually, only through client, database, or collection.
+              collection.read_concern(rc);
+              // Despite what documentation said, it can be set for transactions. Which is nice, as transactions will ignore the
+              // options set on the collection.
+              // TODO: File DOCS ticket to fix the self-contradicting mongocxx documentation.
+              optionsTransaction.read_concern(rc);
+
+              // read_preference
+              if ( rp_string == "PRIMARY" ) {
+                  // TODO: This actually sends primaryPreferred to server. -> File bug in Jira.
+                  // This is benign for this test, but potentially bad for real apps!
+                  // Note that k_secondary works correctly.
+                  rp.mode(mongocxx::read_preference::read_mode::k_primary);
+              }
+              else if ( rp_string == "SECONDARY" ) {
+                  rp.mode(mongocxx::read_preference::read_mode::k_secondary);
+              }
+              else {
+                  BOOST_THROW_EXCEPTION(InvalidConfigurationException("ReadPreference must be PRIMARY or SECONDARY."));
+              }
+              if ( useTransaction ) {
+                  optionsTransaction.read_preference(rp);
+              }
+              else {
+                  optionsFind.read_preference(rp);
+                  optionsAggregate.read_preference(rp);
+              }
+
+          }
+};
+
+void CommitLatency::run() {
+    for (auto&& config : _loop) {
```

Comment:
If you add `using namespace bsoncxx::builder::stream;` to this function, all the `bsoncxx::builder::stream::` can be removed 😄

Reply:
So there's no specific rule for *when* to use `using namespace`..

There are a couple of best practices though.
1. You want to load it at the narrowest block scope to avoid polluting unrelated functions in the same namespace and/or file.
2. If you only need one thing from a namespace frequently, use `using toJson = bsoncxx::to_json;` instead of loading the entire namespace.